### PR TITLE
add nodeSelector to the pods of klusterlet

### PR DESCRIPTION
**Pull request by zhiweiyin318**
story: https://github.com/open-cluster-management/backlog/issues/15911
Signed-off-by: Zhiwei Yin <zyin@redhat.com>

### CODE REVIEW

File: pkg/helpers/helpers.go [hunk 1]
```diff
@@ -505,3 +509,18 @@ func GetComponentNamespace() (string, error) {
 	}
 	return string(nsBytes), nil
 }
+
+func GetNodeSelector(cluster *clusterv1.ManagedCluster) (map[string]string, error) {
+	// This part is to support running pods related local-cluster on specified nodes,like infra nodes.
+	nodeSelector := map[string]string{}
+	if cluster.GetName() == "local-cluster" {
+		annotations := cluster.GetAnnotations()
+		if nodeSelectorString, ok := annotations[AnnotationNodeSelector]; ok {
```
Comment (qiujian16) [inline pkg/helpers/helpers.go:518]: I think a better way is always short circuit.

``` go
if cluster.GetName() != "local-cluster" {
  return nodeSelector
}

annotations := cluster.GetAnnotations()
if annotations == nil {
   return nodeSelector
}

nodeSelectorString, ok := annotations[AnnotationNodeSelector]
if !ok {
  return nodeSelector
}
```

File: pkg/controller/importconfig/manifests/klusterlet/klusterlet.yaml [hunk 2]
```diff
@@ -7,3 +7,12 @@ spec:
   workImagePullSpec: "{{ .WorkImageName }}"
   clusterName: "{{ .ManagedClusterNamespace }}"
   namespace: "{{ .KlusterletNamespace }}"
+  nodePlacement:
+    nodeSelector:
+    {{- range $key, $value := .NodeSelector }}
```
Comment (qiujian16) [inline pkg/controller/importconfig/manifests/klusterlet/klusterlet.yaml:12]: can NodeSelector be nil here?

File: pkg/helpers/helpers.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/helpers/helpers.go]: done

File: pkg/controller/importconfig/manifests/klusterlet/klusterlet.yaml (hunk 2 above)
Comment (zhiweiyin318) [reply pkg/controller/importconfig/manifests/klusterlet/klusterlet.yaml:12]: nil is ok here, but NodeSelector must be defined in config.

**Pull request merged**